File: back_end/109/interview_assessment/ruby_basics/variable_scope/1.rb
# What will the following code print and why? Don't run it until you have tried to answer.

a = 7

def my_value(b)
  b += 10
end

p my_value(a) # 17
puts a # 7

# Local variable 'a' is initialized in the outer scope. 
# We output its value on line 10, which hasn't changed because the reassignment that happens within the method definition 
# doesn't change value of 'a' permanently.

File: books/javascript/functions/1.js
// What does this code log to the console? Does executing the foo function affect the output? Why or why not?

let bar = 1;
function foo() {
  let bar = 2;
}

foo();
console.log(bar); // 1

// JS functions create an inner scope
// The 'bar' variable on line 5 is not the same variable as on line 3
// 'foo()' has no effect on the final output

let bar = 1;
function foo() {
  bar = 2;
}

foo();
console.log(bar); // 2

// Here, 'bar' variable from line 1 is in scope for it to be reassigned within the method
